### PR TITLE
[Group 1] Robot  Code

### PR DESCRIPTION
**Pull request by vincentw921**
Co-authored-by: Kelvin Zhao <t1rbr@users.noreply.github.com>
Co-authored-by: ambers7 <ambers7@users.noreply.github.com>
Co-authored-by: Carmin Vuong <carminvuong@users.noreply.github.com>
Co-authored-by: Samuel Chen <SamChen1738@users.noreply.github.com>

### CODE REVIEW

File: Group1/src/main/java/com/stuypulse/robot/constants/Motors.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ *  - The Current Limit
+ *  - The Open Loop Ramp Rate
+ */
+public interface Motors {
+    public interface Drivetrain {
+        CANSparkMaxConfig left = new CANSparkMaxConfig(true, IdleMode.kBrake, 60, 0);
+        CANSparkMaxConfig right = new CANSparkMaxConfig(false, IdleMode.kBrake, 60, 0);
+    }
+
+    public static class CANSparkMaxConfig {
```
Comment (vincentw921) [inline Group1/src/main/java/com/stuypulse/robot/constants/Motors.java:37]: Add motor config for talon

File: Group1/src/main/java/com/stuypulse/robot/constants/Ports.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+/** This file contains the different ports of motors, solenoids and sensors */
+public interface Ports {
+    public interface Gamepad {
+        int DRIVER = 0;
+        int OPERATOR = 1;
+        int DEBUGGER = 2;
+    }
+    public interface Drivetrain {
+        int LEFT_FRONT = -1;
```
Comment (vincentw921) [inline Group1/src/main/java/com/stuypulse/robot/constants/Ports.java]: Ask engineering about ports

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        CANSparkMax leftBottom = new CANSparkMax(Ports.Drivetrain.LEFT_BOTTOM, MotorType.kBrushless);
+        Motors.Drivetrain.left.configure(leftFront, leftMiddle, leftBottom);
+        left = new MotorControllerGroup(leftFront, leftMiddle, leftBottom);
+        addChild("Left Motor Controller Group", left);
+
+        CANSparkMax rightFront = new CANSparkMax(Ports.Drivetrain.RIGHT_FRONT, MotorType.kBrushless);
+        CANSparkMax rightMiddle = new CANSparkMax(Ports.Drivetrain.RIGHT_MIDDLE, MotorType.kBrushless);
+        CANSparkMax rightBottom = new CANSparkMax(Ports.Drivetrain.RIGHT_BOTTOM, MotorType.kBrushless);
+        Motors.Drivetrain.right.configure(rightFront, rightMiddle, rightBottom);
+        right = new MotorControllerGroup(rightFront, rightMiddle, rightBottom);
```
Comment (vincentw921) [inline Group1/src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java]: change front to top

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+        /** GYRO */
+        navx = new AHRS(SPI.Port.kMXP);
+        addChild("Gyroscope", navx);
+
+        /** ODOMETRY */
+        odometry = new DifferentialDriveOdometry(navx.getRotation2d());
+
+        field = new Field2d();
+        addChild("Field", field);
```
Comment (vincentw921) [inline Group1/src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java:132]: Add drivetrain simulation

File: Group1/src/main/java/com/stuypulse/robot/constants/Motors.java (hunk 1 above)
Comment (selym3) [reply Group1/src/main/java/com/stuypulse/robot/constants/Motors.java:37]: if u guys are using talons yea

File: Group1/src/main/java/com/stuypulse/robot/constants/Ports.java (hunk 2 above)
Comment (selym3) [reply Group1/src/main/java/com/stuypulse/robot/constants/Ports.java]: just make them lol

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java (hunk 3 above)
Comment (selym3) [reply Group1/src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java]: front is good, bottom is not

front/back vs. top/bottom

File: Group1/src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java (hunk 4 above)
Comment (selym3) [reply Group1/src/main/java/com/stuypulse/robot/subsystems/Drivetrain.java:132]: yea -- idk if u should add child the field tho? just see if it works. i feel like u dont want the field in the drivetrain namespace

**Pull request merged**